### DISCUSSION TITLE
Analogmic missing in esp32s2 builds

### DISCUSSION THREAD

**Discussion by eadmaster** [MoonModules/WLED-MM#140] [Q&A] [answered]
i've flashed [this build](https://github.com/MoonModules/WLED/releases/download/v0.14.1-beta.32/WLEDMM_0.14.1-b32.40_esp32s2_PSRAM_M.bin) and there is only digitalmic here:
![digitalmic](https://github.com/MoonModules/WLED/assets/925171/57779538-d9ba-4485-8d90-afaca5c7b88d)

**Comment by eadmaster** [comment]
it's disabled in the code here, i guess for hardware limitations?
https://github.com/MoonModules/WLED/blob/fc173b3bc00694e59b653ca230133052b5476c05/usermods/audioreactive/audio_reactive.h#L1131

**Reply by spiro-c** [reply]
espressif has removed the i2s adc API  from so it is not supported on the new chips 
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/peripherals/i2s.html#overview-of-all-modes

**Comment by softhack007** [comment, marked as answer]
Hi @eadmaster,

It is indeed a hardware limitation. None of the new boards (S2, S3, C3) has support for analog mics any more. Espressif removed the feature from the I2S driver, so we had to remove it from WLED. Sorry that's like it is.
